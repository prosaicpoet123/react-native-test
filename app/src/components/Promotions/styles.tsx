import { StyleSheet } from 'react-native'
import { colours } from '../../config/themes'

const styles = StyleSheet.create({
  promotions: {
    flexDirection: 'column',
    flexWrap: 'wrap',
  },
  promotion: {
    paddingVertical: 2,
    paddingHorizontal: 4,
    borderRadius: 5,
    overflow: 'hidden',
    color: colours.white,
    backgroundColor: colours.pomegranatePrimary,
  },
})

export default styles
