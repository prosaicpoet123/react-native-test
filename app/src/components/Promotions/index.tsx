import React, { FC } from 'react'
import { FlatList, Text, View } from 'react-native'
import styles from './styles'

interface IPromotionsProps {
  promotions: string[]
}

type IProps = IPromotionsProps

const Promotions: FC<IProps> = ({ promotions }) => (
  <View style={styles.promotions}>
    <FlatList
      data={promotions}
      renderItem={({ item }) => <Text style={styles.promotion}>{item}</Text>}
      keyExtractor={(item, index) => `${item}-${index}`}
    />
  </View>
)

export default Promotions
