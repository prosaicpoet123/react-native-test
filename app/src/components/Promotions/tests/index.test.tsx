import React from 'react'
import renderer from 'react-test-renderer'
import Promotions from '../'

describe('<Promotions />', () => {
  it('renders correctly', () => {
    const promotions = ['promotion 1', 'promotion 2']
    const tree = renderer
      .create(<Promotions promotions={promotions} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
