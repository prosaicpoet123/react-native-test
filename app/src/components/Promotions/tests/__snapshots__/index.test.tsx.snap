// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Promotions /> renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "flexWrap": "wrap",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        "promotion 1",
        "promotion 2",
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "backgroundColor": "#F7403A",
              "borderRadius": 20,
              "color": "#FFFFFF",
              "paddingHorizontal": 4,
              "paddingVertical": 2,
            }
          }
        >
          promotion 1
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "backgroundColor": "#F7403A",
              "borderRadius": 20,
              "color": "#FFFFFF",
              "paddingHorizontal": 4,
              "paddingVertical": 2,
            }
          }
        >
          promotion 2
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
